### PR TITLE
Add article on filters/pipes

### PR DESCRIPTION
**Pull request by burkeholland**
Just for you...

### CODE REVIEW

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
+---
+layout: post
+permalink: /angular-2-filters
+title: Creating Filters In Angular 2
+author: burkeholland
+path: 2016-04-02-filters-in-angular-2.md
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: Just a small change to match the correct file name: `2016-08-02-filters-in-angular-2.md`

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    * [Final 1.x Code](#final-1x-code)
+* [Angular 2](#angular-2)
+  * [OrderBy and Filter pipes](#orderby-and-filter-pipes)
+  * [Final 2 code](#final-2-code)
+</div> 
+
+## Angular 1.x
+
+In Angular 1.x, we can create a filter by using the `|` character when we want to transform a paticular value in a template. 
+
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: Small typo "paticular"

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+Filters in Angular 1 are defined simply by putting a `|` on the end of a looping construct - usually `ng-for`. That character - the `|` - is called a "pipe", hence the new name "Pipe" in Angular 2.
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: "usually `ng-for`" change to "typically `ng-repeat` or a single `{{ foo | }}` value. It's common to filter a repeat but also as common to use elsewhere.

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+Filters in Angular 1 are defined simply by putting a `|` on the end of a looping construct - usually `ng-for`. That character - the `|` - is called a "pipe", hence the new name "Pipe" in Angular 2. 
+
+For example, suppose we have a list of groceries, and we want to display each grocery on the page. You might imagine that the controller would look something like this...
+
+{% highlight javascript %}
+const app = {
+  template: `
+    <div>
+      Grocery selected: {{ $ctrl.selectedGrocery.label }}
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: This is a global comment, any instances where you're using `{{` and `}}` curlies, these need escaping like so:

```
{% raw %}{{ todo.label }}{% endraw %}
```

I'll add this to the template and readmes for future help.

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+### OrderBy And Filter
+
+It is also possible to assign filters at the collection level. One of the most powerful filters in Angular 1.x is the `orderBy` filter. Suppose we wanted to show these groceries in alphabetical order. Instead of changing the order of the underlying groceries array, we can just use a filter...
+
+{% highlight javascript %}
+const app = {
+  template: `
+    <div>
+      Grocery selected: {{ $ctrl.selectedGrocery.label | orderBy: 'label' }}
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: I think this: `{{ $ctrl.selectedGrocery.label | orderBy: 'label' }}`

Is meant to go on the `ng-repeat`:

`<li ng-repeat="grocery in $ctrl.groceries | orderBy: 'label'">`

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+      </ul>
+    </div>
+  `,
+  controller() {
+    ...
+  }
+};
+{% endhighlight %}
+
+Elements can also be filtered in a filter. This is another extremely useful feature in Angular 1. In the above example, we might want to filter the list while a user types in a filter box. Angular allows us to pass a model value to the filter which automatically gets applied to the collection.
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: "Elements can also be filtered in a filter." can we change to "Model-bound values can also be passed into filters as arguments."

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+  controller() {
+    ...
+  }
+};
+
+angular
+  .module('app')
+  .component('app', app);
+
+Notice in the above example that multiple filters can be chained together using pipes. Pretty nifty, right? We're just scratching the surface of filters here, but if you've used  Angular 1.x, it's likely that you are already aware of the importance of filters in Angular applications. Filters are incredibly powerful and a good understanding of how to use them will exponentially increase your ability to effectively use Angular in your apps.
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: Nice showing multiple filters in this example.

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+
+### Final 1.x code
+
+<iframe src="https://embed.plnkr.co/2vjOoxbaTiJ2FXpm0jrA/" frameborder="0" border="0" cellspacing="0" cellpadding="0" width="100%" height="250"></iframe>
+
+## Angular 2
+
+As mentioned earlier, **filters are now known as pipes**. They are very similar to how filters in Angular 2 worked, with some serious caveats that we'll discuss here shortly. Angular 2 has built-in pipes just like the filters in Angular 1. For instance, our uppercase filter from the previous example "just works" in Angular 2.
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: I think this needs clarifying: "As mentioned earlier, **filters are now known as pipes**. They are very similar to how filters in Angular 2 worked, with some serious caveats that we'll discuss here shortly." was this meant to say Angular 1?

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 9]
```diff
@@ -0,0 +1,267 @@
+---
+layout: post
+permalink: /angular-2-filters
+title: Creating Filters In Angular 2
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: Can we change the title to "Using filters (pipes) in Angular 2". Emphasis on "using" here as we haven't created a custom one with the `@Pipe()` decorator.

File: _posts/2016-08-02-filters-in-angular-2.md (hunk 5 above)
Comment (burkeholland) [inline _posts/2016-08-02-filters-in-angular-2.md]: Yes - thank you

File: _posts/2016-08-02-filters-in-angular-2.md (hunk 6 above)
Comment (burkeholland) [inline _posts/2016-08-02-filters-in-angular-2.md]: That's not very clear to me. The point I'm trying to make here is that collections can be filtered. How that happens (collection passed to filter) is the abstraction and there's no context for explaining that. What about...

"Collections can also be filtered in Angular 1.x templates using filters"

I know that's redundant, but it's a built in filter called filter. As they say, it is what it is.

Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: Ahhh I see what you mean, I think I misunderstood your "Elements can also be filtered" thinking you meant the `<input>` element. Collections sounds even better, roll with it

File: _posts/2016-08-02-filters-in-angular-2.md [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+  `,
+  controller() {
+    ...
+  }
+};
+{% endhighlight %}
+
+Model-bound values - such as collections, can also be filtered in Angular 1.x using, well, Filters. This is another extremely useful feature in Angular 1. In the above example, we might want to filter the list while a user types in a filter box. Angular allows us to pass a model value to the filter which automatically gets applied to the collection.
+
+const app = {
```
Comment (toddmotto) [inline _posts/2016-08-02-filters-in-angular-2.md]: Just noticed this is missing `{% highlight javascript %}` and `{% endhighlight %}` tags, as well as `{%raw }{{ grocery.label | uppercase }}{% end raw }`

**Pull request merged**